frontend/src/slices/auth/auth.slice.ts

```diff
@@ -36,51 +44,80 @@ const { reducer, actions, name } = createSlice({
-    });
-    builder.addCase(signIn.fulfilled, (state, action) => {
-      state.error = null;
-      state.user = action.payload;
-      state.dataStatus = DataStatus.FULFILLED;
-    });
-    builder.addCase(signIn.rejected, (state, { payload }) => {
-      state.dataStatus = DataStatus.REJECTED;
-      state.error = payload ?? null;
-    });
-    builder.addCase(getCurrent.pending, (state) => {
-      state.dataStatus = DataStatus.PENDING;
-    });
-    builder.addCase(getCurrent.fulfilled, (state, action) => {
-      state.user = action.payload;
-      state.dataStatus = DataStatus.FULFILLED;
-    });
-    builder.addCase(getCurrent.rejected, (state) => {
-      state.dataStatus = DataStatus.REJECTED;
-    });
-    builder.addCase(logOut.pending, (state) => {
-      state.dataStatus = DataStatus.PENDING;
-    });
-    builder.addCase(logOut.fulfilled, (state) => {
-      state.user = initialState.user;
-      state.dataStatus = DataStatus.FULFILLED;
-    });
-    builder.addCase(logOut.rejected, (state, { payload }) => {
-      state.dataStatus = DataStatus.REJECTED;
-      state.error = payload ?? null;
-    });
+    builder
+      .addCase(getCurrent.fulfilled, (state, action) => {
+        state.user = action.payload;
+        state.dataStatus = DataStatus.FULFILLED;
+      })
+      .addCase(getCurrent.rejected, (state) => {
+        state.dataStatus = DataStatus.REJECTED;
+      })
+      .addCase(signIn.fulfilled, (state, action) => {
+        state.error = null;
+        state.user = action.payload;
+        state.dataStatus = DataStatus.FULFILLED;
+      })
+      .addCase(signUp.fulfilled, (state, action) => {
+        state.error = null;
+        state.user = action.payload;
+        state.dataStatus = DataStatus.FULFILLED;
+      })
+      .addCase(editBusiness.fulfilled, (state, action) => {
+        const { firstName, lastName, phone, email, business } = action.payload;
+
+        if (state.user) {
+          const user = state.user as UserEntityObjectWithGroupAndBusinessT;
+          state.user = {
+            ...user,
+            firstName,
+            lastName,
+            phone,
+            email,
+            business: {
+              ...user.business,
+              taxNumber: business.taxNumber,
+              companyName: business.companyName,
+            },
+          };
+        }
+        state.dataStatus = DataStatus.FULFILLED;
+      })
+      .addCase(editCustomer.fulfilled, (state, action) => {
+        const { firstName, lastName, phone, email } = action.payload;
+
+        if (state.user) {
+          state.user = {
+            ...state.user,
+            firstName,
+            lastName,
+            phone,
+            email,
+          };
+        }
+        state.dataStatus = DataStatus.FULFILLED;
+      })
+      .addCase(logOut.fulfilled, (state) => {
+        state.user = initialState.user;
+        state.dataStatus = DataStatus.FULFILLED;
+      })
+      .addMatcher(
+        isAllOf(
```

Comment:
I think there should be an isAnyOf here.
```suggestion
        isAnyOf(
```

Reply:
The same)